conftest.py

```diff
@@ -856,3 +771,136 @@ def num_calls(self) -> int:
     @property
     def details(self) -> t.List[t.Tuple[t.Tuple[t.Any, ...], t.Dict[str, t.Any]]]:
         return self._details
+
+## Reuse database across tests
+
+database_registry = defaultdict(lambda: None)
+
+@pytest.fixture(scope="function")
+def local_experiment(test_dir):
```

Comment:
Totally missed this on the first pass through, but can we add type annotations to the new functions throughout this file?